boost-common/src/main/java/io/openliberty/boost/common/config/BoosterConfigurator.java

```diff
@@ -0,0 +1,107 @@
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.reflections.Reflections;
+
+import io.openliberty.boost.common.BoostException;
+import io.openliberty.boost.common.BoostLoggerI;
+import io.openliberty.boost.common.boosters.AbstractBoosterConfig;
+
+public class BoosterConfigurator {
+
+    /**
+     * take a list of pom boost dependency strings and map to liberty features
+     * for config. return a list of feature configuration objects for each found
+     * dependency.
+     * @param dependencies
+     * @param logger
+     * @return
+     * @throws BoostException
+     * @throws InstantiationException
+     * @throws IllegalAccessException
+     * @throws IllegalArgumentException
+     * @throws InvocationTargetException
+     * @throws NoSuchMethodException
+     * @throws SecurityException
+     */
+    public static List<AbstractBoosterConfig> getBoosterPackConfigurators(Map<String, String> dependencies,
+            BoostLoggerI logger) throws BoostException, InstantiationException, IllegalAccessException, IllegalArgumentException, InvocationTargetException, NoSuchMethodException, SecurityException {
+
+        List<AbstractBoosterConfig> boosterPackConfigList = new ArrayList<AbstractBoosterConfig>();
+        
+        Reflections reflections = new Reflections("io.openliberty.boost.common.boosters");
+
+        Set<Class<? extends AbstractBoosterConfig>> allClasses = reflections.getSubTypesOf(AbstractBoosterConfig.class);
+        for(Class<? extends AbstractBoosterConfig> boosterClass : allClasses) {
+            if (dependencies.containsKey(AbstractBoosterConfig.getCoordindates(boosterClass))) {
+                Constructor<?> cons = boosterClass.getConstructor(Map.class, BoostLoggerI.class);
+                Object o = cons.newInstance(dependencies, logger);
+                if(o instanceof AbstractBoosterConfig) {
+                    boosterPackConfigList.add((AbstractBoosterConfig)o);
+                }
+                else {
+                    throw new BoostException("Found a class in io.openliberty.boost.common.boosters that did not extend AbstractBoosterConfig. This should never happen.");
+                }
+            }
+        }
+        
+        return boosterPackConfigList;
+    }
+
+    public static void generateLibertyServerConfig(String libertyServerPath,
+            List<AbstractBoosterConfig> boosterPackConfigurators, String warName, BoostLoggerI logger) throws Exception {
+
+        LibertyServerConfigGenerator serverConfig = new LibertyServerConfigGenerator(libertyServerPath, logger);
+
+        // Loop through configuration objects and get features and XML config
+        // (if any)
+        for (AbstractBoosterConfig configurator : boosterPackConfigurators) {
+            serverConfig.addFeature(configurator.getFeature());
+            serverConfig.addBoosterConfig(configurator);
+        }
```

Comment:
One of the items I was working on for the mpConfig support came from a comment about collapsing these two separate calls into one action - as they both are adding to the server.xml. addFeature is a local method that adds the feature string retrieved by the configurator to the server.xml doc object as a feature stanza in the featureManager block. addBoosterConfig is a local method that takes a configurator object and in turn calls that configurator object with a reference to the same doc object to be updated with that configurator's config stanzas. Should both the configurator's feature name and its config stanzas (if there are any) be written out at the same time in a single call?

Reply:
Sorry I'm a bit confused. I don't believed I changed the implementation of this method (it's showing up as additions because git didn't pick up that this file had been renamed). I'm not sure if I appreciate the difference in combining the two calls vs keeping them separate...but from my understanding I don't see a problem the way it is now (one call to add the feature, and another to add additional config). Although maybe `addBoosterConfig` should be taking in `configurator.getServerProperties()` instead of the entire `configurator` object.

Anyways not sure how much this relates to my PR, did you have any changes that you wanted me to make here?

Reply:
you're correct - it doesn't have anything to do with the changes you made - i was thinking proactively and meant this more of a point of discussion. I should have made it is a comment vs a review to get your thoughts. Your response makes sense, I can mark it resolved.

Reply:
Sure, no worries :)